File: samoa/server/context.cpp
#include "samoa/server/context.hpp"
#include <boost/python.hpp>

namespace server {

using namespace boost::python;

void make_context_bindings()
{
    class_<context, context::ptr_t, boost::noncopyable>(
        "context", init<core::proactor::ptr_t>())
        .add_property("proactor", make_function(&context::get_proactor,
            return_value_policy<copy_const_reference>()));
}

}

File: py_src/samoa/server/context.py

import getty
import samoa.core.protobuf

import _server

class Context(_server.Context):

    @getty.requires(
        cluster_state = samoa.core.protobuf.ClusterState)
    def __init__(self, cluster_state):
        _server.Context.__init__(self, cluster_state)
        self.spawn_tasklets()

